### PR TITLE
Shail fix parameterisation ordering

### PR DESCRIPTION
**Pull request by shaildesai96**
## Linked Issues

Does this PR close or fix any Issues? Remember to create an Issue before starting work so that your fix / proposal can be addressed by the team.

e.g. Closes #{ID}

## Description

Gets wires 'pointing' in same direction for PictureFrame shapes

## Interface Changes

If you've had to update an interface or introduce a new interface as part of your change then let us know here.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/geometry/parameterisations.py [hunk 1]
```diff
@@ -1655,7 +1661,7 @@ def create_shape(self, label=""):
         # Bottom leg is flat
 
         bot_leg = PictureFrameTools._make_flat_leg(
-            axis,
+            [0, 1, 0],
```
Comment (je-cook) [inline bluemira/geometry/parameterisations.py:1664]: Whats the reasoning behind having these constant instead of using axis?

**Review by je-cook — COMMENTED** [review]
Just a few little things as I was passing

File: bluemira/geometry/parameterisations.py (hunk 1 above)
Comment (shaildesai96) [reply bluemira/geometry/parameterisations.py:1664]: the way the angles are defined - clockwise vs counterclockwise. CP coils have curves in bothe directions, so need to specify fo at least one

File: examples/design/EU-DEMO/EUDEMO_reactor.py [hunk 2]
```diff
@@ -219,20 +219,13 @@
     },
     "TF Coils": {
         "runmode": "run",  # ["run", "read", "mock"]
-        "param_class": "TripleArc",
+        "param_class": "PictureFrame",
```
[the remark this answers is not in the document]
Comment (shaildesai96) [reply examples/design/EU-DEMO/EUDEMO_reactor.py:222]: yup forgot

**Comment by CoronelBuendia** [comment]
```python
from bluemira.geometry.parameterisations import PrincetonD, TaperedPictureFrame, PictureFrame, TopDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame

from bluemira.geometry.tools import sweep_shape, make_polygon

c = [PrincetonD, TaperedPictureFrame, PictureFrame, TopDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame]

results = []
for cls in c:
    try:
        path = cls({"x1":{"value": 2}}).create_shape()
        profile = make_polygon({"x": [1.75, 2.25, 2.25, 1.75][::-1], "y": [-0.25, -0.25, 0.25, 0.25][::-1], "z": 0})
        results.append([profile, path])
    except:
        print(cls.__name__)
```
```python
+-------------------------------------------------------------------------+
| WARNING: BoundedVariable 'x1': value was set to above its upper         |
| bound. Adjusting bound.                                                 |
+-------------------------------------------------------------------------+
+-------------------------------------------------------------------------+
| WARNING: BoundedVariable 'x1': value was set to below its lower         |
| bound. Adjusting bound.                                                 |
+-------------------------------------------------------------------------+
TaperedPictureFrame
TopDomeFlatInnerCurvedPictureFrame
FullDomeFlatInnerCurvedPictureFrame
FullDomeFlatInnerCurvedPictureFrame
```

**Comment by CoronelBuendia** [comment]
> ```python
> from bluemira.geometry.parameterisations import PrincetonD, TaperedPictureFrame, PictureFrame, TopDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame
> 
> from bluemira.geometry.tools import sweep_shape, make_polygon
> 
> c = [PrincetonD, TaperedPictureFrame, PictureFrame, TopDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, FullDomeFlatInnerCurvedPictureFrame]
> 
> results = []
> for cls in c:
>     try:
>         path = cls({"x1":{"value": 2}}).create_shape()
>         profile = make_polygon({"x": [1.75, 2.25, 2.25, 1.75][::-1], "y": [-0.25, -0.25, 0.25, 0.25][::-1], "z": 0})
>         results.append([profile, path])
>     except:
>         print(cls.__name__)
> ```
> 
> ```python
> +-------------------------------------------------------------------------+
> | WARNING: BoundedVariable 'x1': value was set to above its upper         |
> | bound. Adjusting bound.                                                 |
> +-------------------------------------------------------------------------+
> +-------------------------------------------------------------------------+
> | WARNING: BoundedVariable 'x1': value was set to below its lower         |
> | bound. Adjusting bound.                                                 |
> +-------------------------------------------------------------------------+
> TaperedPictureFrame
> TopDomeFlatInnerCurvedPictureFrame
> FullDomeFlatInnerCurvedPictureFrame
> FullDomeFlatInnerCurvedPictureFrame
> ```

OK so part of this is that most of the picture frames don't have x1 as a variable name... This isn't a problem per se, but it could get a bit annoying given their predominant use case.

The first one fails with an error I have never seen before.

File: tests/bluemira/geometry/test_parameterisations.py [hunk 3]
```diff
@@ -169,6 +170,20 @@ def test_no_corners(self):
...
+                "x2": {"value": 16},
+                "z1": {"value": 8},
+                "z2": {"value": -8},
+                "ri": {"value": 1, "upper_bound": 1},
+                "ro": {"value": 1},
+            }
+        )
+        wire = p.create_shape()
+        assert _wire_edges_tangent(wire._shape)
+
```
Comment (CoronelBuendia) [inline tests/bluemira/geometry/test_parameterisations.py:186]: Test all picture frame parameterisations please

**Comment by shaildesai96** [comment]
As per discussions, I am getting rid of parmeterisation of CurvedPictureframe shapes with tapered centreposts, as the wire parameterisation is not planned to be used in the TF coil build procedure

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]
We should agree on how to document the variables of a given parameterisation. At present the docstrings you have written do not correspond to the signatures of these classes, and as such don't really make sense.

EDIT: See #608 

Elsewhere we've gone with using the description field of the BoundedVariables. Perhaps @je-cook could come up with a decent way of doing this for all GeometryParameterisations

**Comment by CoronelBuendia** [comment]
A few code style and misc comments. I've given this a bit of poking:

(NOTE: TaperedPictureFrame fails with lack of tangency, @shaildesai96  is this expected?)

```python
from bluemira.geometry.tools import sweep_shape, make_polygon
from bluemira.geometry.parameterisations import PictureFrame, TaperedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, BotDomeFlatInnerCurvedPictureFrame, TopDomeFlatInnerCurvedPictureFrame
from copy import deepcopy
import numpy as np
from bluemira.display import show_cad

dr, dy = 0.2, 0.4
xs = make_polygon({"x":[-dr, dr, dr, -dr], "y":[-dy, -dy, dy, dy], "z":0}, closed=True)

shapes  = []
results = []
offset = 0
for p in [PictureFrame, TaperedPictureFrame, FullDomeFlatInnerCurvedPictureFrame, BotDomeFlatInnerCurvedPictureFrame, TopDomeFlatInnerCurvedPictureFrame]:
    cross_section = deepcopy(xs)
    shape = p().create_shape()
    shapes.append(shape)
    x_min = np.min(shape.discretize(ndiscr=100, byedges=True).x)
    cross_section.translate((x_min, 0, 0))
    try:
        result = sweep_shape(cross_section, shape)
        result.translate((0, offset, 0))
        results.append(result)
    except:
        print(p.__name__)
        pass
    offset += 3
```

In particular because I was worried how some of these might sweep.

image

It's as we discussed before; if the bending radius is lesser than the half-width (radial) of the WP, you get overlaps which manifest as voids, and you get some invalid solids effectively.

With dr=0.01 here:
image

**Comment by je-cook** [comment]
> Elsewhere we've gone with using the description field of the BoundedVariables. Perhaps @je-cook could come up with a decent way of doing this for all GeometryParameterisations

So you want to include in the docstring what variables the parameterisation uses automatically yeah?

I'll have a think

File: bluemira/geometry/parameterisations.py [hunk 4]
```diff
@@ -999,57 +1001,58 @@ def _make_domed_leg(
...
         z_mid,
+        r_c,
         flip=False,
-        r_c=0.0,
     ):
         """
         Makes smooth dome for CP coils. This includes a initial straight section
         and a main curved dome section, with a transitioning 'joint' between them,
-        producing smooth tangent curves.
+        producing smooth tanent curves.
```
[the remark this answers is not in the document]
Comment (shaildesai96) [reply bluemira/geometry/parameterisations.py]: Done

**Comment by CoronelBuendia** [comment]
> > Elsewhere we've gone with using the description field of the BoundedVariables. Perhaps @je-cook could come up with a decent way of doing this for all GeometryParameterisations
> 
> So you want to include in the docstring what variables the parameterisation uses automatically yeah?
> 
> I'll have a think

Doesn't need to be automatic, just that the variables are not actually inputs to the GeometryParameterisation per se, but are worth documenting in the docstrings.

**Comment by je-cook** [comment]
Ok I sort of have one for you @CoronelBuendia. If instead of having var_dict as the input we have the actual bounded variable names (with their defaults set like below) then the only change to the interface would be the addition of `**` and it would then be documented in the docstring. We could use named tuples directly in the defaults to make them less opaque. Just a tuple could be used if the user wants. As written below it has to be keyword arguments used which would enforce the dictionary input more.

eg 
```python
class opt:
    def __init__(self, *, x1=BoundedVariable('x1', 4, 2, 6), x2=BoundedVariable('x2', 14, 10, 18), dz=BoundedVariable('dz', 0, -0.5, 0.5)):
        variables = OptVariables(
            [
                BoundedVariable(
                    "x1", 4, lower_bound=2, upper_bound=6, descr="Inboard limb radius"
                ),
                BoundedVariable(
                    "x2",
                    14,
                    lower_bound=10,
                    upper_bound=18,
                    descr="Outboard limb radius",
                ),
                BoundedVariable(
                    "dz",
                    0,
                    lower_bound=-0.5,
                    upper_bound=0.5,
                    descr="Vertical offset from z=0",
                ),
            ],
            frozen=True,
        )
        var_dict = dict(x1, x2,dz) # this is sudo code but you get the idea
        variables.adjust_variables(var_dict, strict_bounds=False)


# Use
parameterisation = opt(**var_dict)

**Comment by CoronelBuendia** [comment]
> Ok I sort of have one for you @CoronelBuendia. If instead of having var_dict as the input we have the actual bounded variable names (with their defaults set like below) then the only change to the interface would be the addition of `**` and it would then be documented in the docstring. We could use named tuples directly in the defaults to make them less opaque. Just a tuple could be used if the user wants. As written below it has to be keyword arguments used which would enforce the dictionary input more.
> 
> eg
> 
> ```python
> class opt:
>     def __init__(self, *, x1=BoundedVariable('x1', 4, 2, 6), x2=BoundedVariable('x2', 14, 10, 18), dz=BoundedVariable('dz', 0, -0.5, 0.5)):
>         variables = OptVariables(
>             [
>                 BoundedVariable(
>                     "x1", 4, lower_bound=2, upper_bound=6, descr="Inboard limb radius"
>                 ),
>                 BoundedVariable(
>                     "x2",
>                     14,
>                     lower_bound=10,
>                     upper_bound=18,
>                     descr="Outboard limb radius",
>                 ),
>                 BoundedVariable(
>                     "dz",
>                     0,
>                     lower_bound=-0.5,
>                     upper_bound=0.5,
>                     descr="Vertical offset from z=0",
>                 ),
>             ],
>             frozen=True,
>         )
>         var_dict = dict(x1, x2,dz) # this is sudo code but you get the idea
>         variables.adjust_variables(var_dict, strict_bounds=False)
> 
> 
> # Use
> parameterisation = opt(**var_dict)
> ```

Don't want to change the API in this PR (there's already an issue sort of related to that), open to doing it in another. Right now the docstrings here are not valid. Maybe `Notes` or something.

**Comment by je-cook** [comment]
Yeah a Notes section would cover it

**Review by je-cook — CHANGES_REQUESTED** [review]
A few minor changes

File: bluemira/geometry/parameterisations.py [hunk 5]
```diff
@@ -1432,19 +1471,19 @@ def create_shape(self, label=""):
...
+    Curved picture-frame geometry parameterisation, with
+    top flat and bottom domed. Straight (non-tapered) inner limb
 
     Parameters
     ----------
     var_dict: Optional[dict]
         Dictionary with which to update the default values of the parameterisation.
     axis: np.ndarray
-            [x,y,z] vector normal to plane of parameterisation
+        [x,y,z] vector normal to plane of parameterisation
```
[the remark this answers is not in the document]
Comment (shaildesai96) [reply bluemira/geometry/parameterisations.py:1485]: It's an input to make_domed_leg

[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/geometry/parameterisations.py:1485]: oh ok  :+1:

File: bluemira/geometry/parameterisations.py [hunk 6]
```diff
@@ -1641,44 +1690,49 @@ def create_shape(self, label=""):
...
 
     Parameters
     ----------
+
+    Notes
+    -----
+    The dictionary keys in var_dict are:
+
     var_dict: Optional[dict]
         Dictionary with which to update the default values of the parameterisation.
```
Comment (je-cook) [inline bluemira/geometry/parameterisations.py:1733]:
```suggestion
    var_dict: Optional[dict]
        Dictionary with which to update the default values of the parameterisation.
```

**Review by je-cook — APPROVED** [review]

File: bluemira/geometry/parameterisations.py (hunk 6 above)
Comment (je-cook) [inline bluemira/geometry/parameterisations.py]:
```suggestion
    var_dict: Optional[dict]
        Dictionary with which to update the default values of the parameterisation.
        
    Notes
    -----
    The dictionary keys in var_dict are:

```

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]
Docstrings of these classes still not representative I'm afraid. Please ad a `Notes\n-----` section in to describe the variables rather than put them under `Parameters\n-----------`

**Review by CoronelBuendia — APPROVED** [review]

**Pull request merged**